pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,353 @@ def Sphere(
+
+    radius_resolution : int, default: 5
+        Number of points in radial direction.
+
+    start_theta : float, default: 0.0
+        Starting longitude angle in degrees.
+
+    end_theta : float, default: 360.0
+        Ending longitude angle in degrees.
+
+    theta_resolution : int, default: 30
+        Number of points in theta direction.
+
+    start_phi : float, default: 0.0
+        Starting latitude angle in degrees.
+
+    end_phi : float, default: 180.0
+        Ending latitude angle in degrees.
+
+    phi_resolution : int, default: 30
+        Number of points in phi direction,
+        inclusive of polar axis if applicable.
+
+    radius : sequence[float], optional
+        A monotonically increasing sequence of values specifying radial
+        points. If specified, it overrides any choices for ``outer_radius``,
+        ``inner_radius``, and ``radius_resolution``.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying theta
+        points in degrees. If specified, it overrides any choices for
+        ``start_theta``, ``end_theta``, and ``theta_resolution``.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying phi
+        points in degrees. If specified, it overrides any choices for
+        ``start_phi``, ``end_phi``, and ``phi_resolution``.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees latitude.
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.Sphere
+
+    Examples
+    --------
+    Create a solid sphere.
+
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> solid_sphere = pv.SolidSphere()
+    >>> solid_sphere.plot(show_edges=True)
+
+    A solid sphere is 3D in comparison to the 2d `pyvista.Sphere`.
+    Generate a solid hemi-sphere to see the internal structure.
+
+    >>> isinstance(solid_sphere, pv.UnstructuredGrid)
+    True
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180, end_theta=360
+    ... )
+    >>> partial_solid_sphere.plot(show_edges=True)
+
+    To see the cell structure inside the solid sphere,
+    only 1/4 of the sphere is generated. The cells exploded
+    and colored by radial position.
+
+    >>> partial_solid_sphere = pv.SolidSphere(
+    ...     start_theta=180,
+    ...     end_theta=360,
+    ...     start_phi=0,
+    ...     end_phi=90,
+    ...     radius_resolution=5,
+    ...     theta_resolution=8,
+    ...     phi_resolution=8,
+    ... )
+    >>> partial_solid_sphere["cell_radial_pos"] = np.linalg.norm(
+    ...     partial_solid_sphere.cell_centers().points, axis=-1
+    ... )
+    >>> partial_solid_sphere.explode(1).plot()
```

Comment:
One more remark: I wonder if it would be worth adding an example that generates an angular grid that leads to close to uniform cell surfaces. See e.g. [Sphere Point Picking at wolfram](https://mathworld.wolfram.com/SpherePointPicking.html): using a uniform polar angle grid leads to smaller faces near the poles. This is also visible in the first screenshot of the PR comment:

<img src="https://github.com/pyvista/pyvista/assets/39341281/189f7e1b-ed31-431f-aaba-c6680b489a2d" width="480">

Generating `phi = np.rad2deg(np.arccos(np.linspace(1, -1, n_phi)))` should give faces of similar size I think.

Reply:
I like this idea and we don't currently have an example using the really cool feature of using a custom sequence. 

Slightly off topic: For sampling purposes, we can use my own personal project pyransame, which gives you uniform sampling no matter how unevenly distributed your mesh is.

Reply:
This is done.

Linear sampling of phi
![image](https://github.com/pyvista/pyvista/assets/39341281/c9c531b7-178e-4636-a33d-5f9fd86ff548)

Using your suggestion for phi
![image](https://github.com/pyvista/pyvista/assets/39341281/e1b24fbc-c2ca-4339-9271-cf7156c1d129)

Reply:
For something so featureless, this looks awesome, thanks :smile: